mediathread/djangosherd/models.py

```diff
@@ -216,16 +211,24 @@ def global_annotation(self, asset, author, auto_create=True):
         """
         args = NULL_FIELDS.copy()
         args.update(asset=asset, author=author)
+        created = False
 
         if auto_create:
-            return self.get_or_create(**args)
+            gannotation, created = self.get_or_create(**args)
         else:
             try:
                 gannotation = self.get(**args)
             except SherdNote.DoesNotExist:
                 gannotation = None
+
             return gannotation, False
 
+        if gannotation and not gannotation.is_global_annotation:
```

Comment:
I think setting the flag here might be unnecessary if you add the `is_global_annotation` to the args above, e.g. `args.update(is_global_annotation=True)`. This also might prevent some oddness in the `get_or_create` at Ln. 217. The `get_or_create` used to retrieve a unique annotation for an asset/user combination by  checking for an empty title and empty range. As written now, it's retrieving a unique annotation for an asset/user combination simply with an empty title. That might not be quite as unique.

Reply:
Yeah, I think you're right. I've made that change in a new commit here.